cmd/container_clone.go

```diff
@@ -26,148 +28,77 @@ var containerCloneCmd = &cobra.Command{
-		}
-
-		if targetEnvironmentName == "" {
-			// use same env name as the source env
-			targetEnvironmentName = sourceEnvironment.Name
-		}
-
-		targetEnvironment := utils.FindByEnvironmentName(environments.GetResults(), targetEnvironmentName)
-
-		if targetEnvironment == nil {
-			utils.PrintlnError(fmt.Errorf("environment %s not found", targetEnvironmentName))
-			utils.PrintlnInfo("You can list all environments with: qovery environment list")
-			os.Exit(1)
+			if err != nil {
+				utils.PrintlnError(err)
+				os.Exit(1)
+				panic("unreachable") // staticcheck false positive: https://staticcheck.io/docs/checks#SA5011
+			}
 		}
 
-		var storage []qovery.ServiceStorageRequestStorageInner
+		targetEnvironmentId := envId // use same env as the source env
+		if targetEnvironmentName != "" {
 
-		for _, s := range container.Storage {
-			storage = append(storage, qovery.ServiceStorageRequestStorageInner{
-				Type:       s.Type,
-				Size:       s.Size,
-				MountPoint: s.MountPoint,
-			})
-		}
+			targetEnvironmentId, err = getEnvironmentContextResourceId(client, targetEnvironmentName, targetProjectId)
 
-		var ports []qovery.ServicePortRequestPortsInner
-
-		for _, p := range container.Ports {
-			ports = append(ports, qovery.ServicePortRequestPortsInner{
-				Name:               p.Name,
-				InternalPort:       p.InternalPort,
-				ExternalPort:       p.ExternalPort,
-				PubliclyAccessible: p.PubliclyAccessible,
-				IsDefault:          p.IsDefault,
-				Protocol:           &p.Protocol,
-			})
+			if err != nil {
+				utils.PrintlnError(err)
+				os.Exit(1)
+				panic("unreachable") // staticcheck false positive: https://staticcheck.io/docs/checks#SA5011
+			}
 		}
 
 		if targetContainerName == "" {
+			// use same container name as the source container
 			targetContainerName = container.Name
 		}
 
-		req := qovery.ContainerRequest{
-			Storage:             storage,
-			Ports:               ports,
-			Name:                targetContainerName,
-			Description:         container.Description,
-			RegistryId:          container.Registry.Id,
-			ImageName:           container.ImageName,
-			Tag:                 container.Tag,
-			Arguments:           container.Arguments,
-			Entrypoint:          container.Entrypoint,
-			Cpu:                 &container.Cpu,
-			Memory:              &container.Memory,
-			MinRunningInstances: &container.MinRunningInstances,
-			MaxRunningInstances: &container.MaxRunningInstances,
-			AutoPreview:         &container.AutoPreview,
-			Healthchecks:        container.Healthchecks,
+		req := qovery.CloneContainerRequest{
+			Name:          targetContainerName,
+			EnvironmentId: targetEnvironmentId,
 		}
 
-		createdService, res, err := client.ContainersApi.CreateContainer(context.Background(), targetEnvironment.Id).ContainerRequest(req).Execute()
+		clonedService, res, err := client.ContainersApi.CloneContainer(context.Background(), container.Id).CloneContainerRequest(req).Execute()
```

Comment:
_nitpick_: maybe we can replace service with container here:  `clonedService` => `clonedContainer`